Day0/Reetwiz/assignment1/render_ppm.c

```diff
@@ -0,0 +1,155 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct
+{
+    unsigned int cx, cy, r;
+} CIRCLE;
+
+typedef struct
+{
+    unsigned int cx, cy, width, height;
+} RECT;
+
+typedef struct
+{
+    unsigned char R, G, B;
+} PIX;
+
+typedef struct
+{
+    unsigned int row, column;
+    PIX **pixel; //pixels stored in 2d array of rows and columns
+} IMG;
+
+IMG createLayout(unsigned int row, unsigned int col)
+{
+    IMG ppm_layout;
+    ppm_layout.row = row;
+    ppm_layout.column = col;
+   
+    ppm_layout.pixel = (PIX **)malloc(sizeof(PIX *) * ppm_layout.row);
+    if (!ppm_layout.pixel)
+    {
+        printf("cannot allocate memory");
+        exit(1);
+    }
+
+    for (int i = 0; i < ppm_layout.row; ++i)
+    {
+        *(ppm_layout.pixel + i) = (PIX *)malloc(sizeof(PIX *) * ppm_layout.column);
+        if (!(*(ppm_layout.pixel + i)))
+        {
+            printf("cannot allocate memory");
+            exit(1);
+        }
+    }
+    return ppm_layout;
+}
+
+IMG BlackFill(IMG ppm_layout)
```

Comment:
There is not point in returning the IMG, since IMG points to some memory, the changes will be reflected